File: app/src/main/java/com/yaros/kitchen/api/AuthInterceptor.java
package com.yaros.kitchen.api;

import java.io.IOException;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class AuthInterceptor implements Interceptor {

    private String waiterToken;

    public AuthInterceptor(String apiKey) {
        waiterToken = apiKey;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        if (waiterToken!=null) {
            HttpUrl url = chain.request().url()
                    .newBuilder()
                    .addQueryParameter("waiter_token", waiterToken)
                    .build();
            Request request = chain.request().newBuilder().url(url).build();
            return chain.proceed(request);
        }
        else
            return chain.proceed(chain.request());
    }
}

File: app/src/main/java/com/yaros/kitchen/api/AuthInterceptor.kt
package com.yaros.kitchen.api

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class AuthInterceptor(private val waiterToken: String?) : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        return if (waiterToken != null) {
            val url = chain.request().url()
                .newBuilder()
                .addQueryParameter("waiter_token", waiterToken)
                .build()
            val request = chain.request().newBuilder().url(url).build()
            chain.proceed(request)
        } else chain.proceed(chain.request())
    }




}